editors/code/package.json

```diff
@@ -39,24 +39,24 @@
         "vscode-languageclient": "7.0.0-next.1"
     },
     "devDependencies": {
-        "@rollup/plugin-commonjs": "^12.0.0",
-        "@rollup/plugin-node-resolve": "^8.0.0",
-        "@types/glob": "^7.1.1",
+        "@rollup/plugin-commonjs": "^13.0.0",
+        "@rollup/plugin-node-resolve": "^8.1.0",
+        "@types/glob": "^7.1.2",
         "@types/mocha": "^7.0.2",
-        "@types/node": "^14.0.5",
+        "@types/node": "^12.12.0",
```

Comment:
Well why, you have the point, but my point was that `^`-character prefix in semver version (`^12.8.0`) in `package.json` still pulled `@types/node@12.12.0` because `^` [allows selection of minor and patch versions greater than the one specified](https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004). If current vscode uses `nodejs12.8` than we better use the same minor version by specifying:

```suggestion
        "@types/node": "12.8.x"
```
Packages of `@types/*` namespace have [specific rules about this](https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-definitely-typed-package-versions-relate-to-versions-of-the-corresponding-library), having read them I can conclude that we have to ensure that the major version of `@types/foo` package is equal to the major version of `foo` library itself and the minor version of `@types/foo` has to be less-than-or-equal to the version `foo` library itself (in our case *the* `foo` library is nodejs stdlib)
We cannot just use the newer minor version of `@types/*` if we are using the lower minor version of the runtime library (which is nodejs stdlib in this case) in our code.
For example let say we have
```jsonc
// package.json
{
    "@types/nodejs": "^12.8.0",
}
// package-lock.json
{
     "@types/nodejs": "12.12.0", // boom, npm has selected greater minor version in its lockfile due to ^
}
```
```ts
import stream from "stream";
stream.foo();
// But stream.foo() was added in nodejs.12.12, and is not present in nodejs.12.8
// however, the typescript compiler doesn't issue an error, since the `@types/nodejs` package
// includes the declaration for `stream.foo()`, because we have `@types/nodejs@12.12` in our lock file
// and this will be a runtime error when we hit this line of code
```

Reply:
Fair point. Switched to `~12.7.0` because there's no `~12.8.0`.